diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0c11507838..8dc2391f00 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -528,7 +528,7 @@ endif(NOT MSVC)
 # force it to do mobile build with host toolchain - which is useful for testing
 # purpose.
 if(ANDROID OR IOS OR DEFINED ENV{BUILD_PYTORCH_MOBILE_WITH_HOST_TOOLCHAIN})
-  set(INTERN_BUILD_MOBILE ON)
+  #set(INTERN_BUILD_MOBILE ON)
 
   if(DEFINED ENV{BUILD_PYTORCH_MOBILE_WITH_HOST_TOOLCHAIN})
     # C10_MOBILE is derived from Android/iOS toolchain macros in
diff --git a/cmake/ProtoBuf.cmake b/cmake/ProtoBuf.cmake
index 8d7633c4ab..2d859461d4 100644
--- a/cmake/ProtoBuf.cmake
+++ b/cmake/ProtoBuf.cmake
@@ -84,7 +84,7 @@ endmacro()
 # Main entry for protobuf. If we are building on Android, iOS or we have hard
 # coded BUILD_CUSTOM_PROTOBUF, we will hard code the use of custom protobuf
 # in the submodule.
-if(ANDROID OR IOS)
+if((ANDROID OR IOS) AND BUILD_CUSTOM_PROTOBUF)
   if(NOT BUILD_CUSTOM_PROTOBUF)
     message(WARNING
         "For Android and iOS cross compilation, I am automatically using "
